MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/labs/RecyclerViewOnMapActivity.java

```diff
@@ -0,0 +1,275 @@
+import com.mapbox.mapboxsdk.geometry.LatLng;
+import com.mapbox.mapboxsdk.maps.MapView;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Use a recyclerview with a Mapbox map to easily explore content all on one screen
+ */
+public class RecyclerViewOnMapActivity extends AppCompatActivity {
+
+  private MapView mapView;
+  public MapboxMap mapboxMap;
+  private RecyclerView recyclerView;
+  private LocationRecyclerViewAdapter locationAdapter;
+  private ArrayList<SingleRecyclerViewLocation> locationList;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    // Mapbox access token is configured here. This needs to be called either in your application
+    // object or in the same activity which contains the mapview.
+    Mapbox.getInstance(this, getString(R.string.access_token));
+
+    // This contains the MapView in XML and needs to be called after the access token is configured.
+    setContentView(R.layout.activity_lab_recycler_view_on_map);
+
+    recyclerView = (RecyclerView) findViewById(R.id.rv_on_top_of_map);
+
+    Toast.makeText(this, R.string.toast_instruction, Toast.LENGTH_SHORT).show();
+
+    // Initialize the map view
+    mapView = (MapView) findViewById(R.id.mapView);
+    mapView.onCreate(savedInstanceState);
+    mapView.getMapAsync(new OnMapReadyCallback() {
+      @Override
+      public void onMapReady(MapboxMap mapboxMap) {
+        RecyclerViewOnMapActivity.this.mapboxMap = mapboxMap;
+
+        setUpLists();
+
+        // Set up the recyclerView
+        locationAdapter = new LocationRecyclerViewAdapter(locationList, mapboxMap);
+        recyclerView.setLayoutManager(new LinearLayoutManager(getApplicationContext(),
+            LinearLayoutManager.HORIZONTAL, true));
+        recyclerView.setItemAnimator(new DefaultItemAnimator());
+        recyclerView.setAdapter(locationAdapter);
+      }
+    });
+
+  }
+
+  // Add the mapView lifecycle to the activity's lifecycle methods
+  @Override
+  public void onResume() {
+    super.onResume();
+    mapView.onResume();
+  }
+
+  @Override
+  protected void onStart() {
+    super.onStart();
+    mapView.onStart();
+  }
+
+  @Override
+  protected void onStop() {
+    super.onStop();
+    mapView.onStop();
+  }
+
+  @Override
+  public void onPause() {
+    super.onPause();
+    mapView.onPause();
+  }
+
+  @Override
+  public void onLowMemory() {
+    super.onLowMemory();
+    mapView.onLowMemory();
+  }
+
+  @Override
+  protected void onDestroy() {
+    super.onDestroy();
+    mapView.onDestroy();
+  }
+
+  @Override
+  protected void onSaveInstanceState(Bundle outState) {
+    super.onSaveInstanceState(outState);
+    mapView.onSaveInstanceState(outState);
+  }
+
+  public boolean onOptionsItemSelected(MenuItem item) {
+    switch (item.getItemId()) {
+      case android.R.id.home:
```

Comment:
The parent/child relationship is [already set up](https://github.com/mapbox/mapbox-android-demo/pull/400/files#diff-d5babffa5e8bb42bd1d56e07c65c4c63R176) in `AndroidManifest.xml` for all of the examples, correct. This handles closing the example activity and returning to `MainActivity`.

However, I'm doing `onOptionsItemSelected` to deal with returning to _the same_ nav drawer section that the example lives in, rather than always returning to the `Getting Started` section when the user closes the example. I haven't figured out another way to deal with this navigation, other than by adding the `onOptionsItemSelected` method to each example. Is there something to add in `AndroidManifest.xml` which can accomplish this?   cc @tobrun       

These comments are also applicable to [this comment chain in the other demo app pr](https://github.com/mapbox/mapbox-android-demo/pull/402#discussion_r126604210)